File: trip-kit-booking/src/main/java/com/skedgo/tripkit/booking/AuthProvider.java
package com.skedgo.tripkit.booking;

import androidx.annotation.Nullable;

import com.google.gson.annotations.JsonAdapter;

import static org.immutables.gson.Gson.TypeAdapters;
import static org.immutables.value.Value.Immutable;
import static org.immutables.value.Value.Style;

@TypeAdapters
@Immutable
@Style(passAnnotations = JsonAdapter.class)
@JsonAdapter(GsonAdaptersAuthProvider.class)
public interface AuthProvider {
  @Nullable String modeIdentifier();
  @Nullable String provider();
  @Nullable String action();
  @Nullable String url();
  @Nullable String actionTitle();
  @Nullable String status();
  @Nullable CompanyInfo companyInfo();
}

File: trip-kit-booking/src/main/java/com/skedgo/tripkit/booking/AuthProvider.kt
package com.skedgo.tripkit.booking

import com.google.gson.annotations.JsonAdapter
import org.immutables.gson.Gson.TypeAdapters
import org.immutables.value.Value.Immutable
import org.immutables.value.Value.Style

@TypeAdapters
@Immutable
@Style(passAnnotations = [JsonAdapter::class])
@JsonAdapter(
    GsonAdaptersAuthProvider::class
)
interface AuthProvider {
    fun modeIdentifier(): String?

    fun provider(): String?

    fun action(): String?

    fun url(): String?

    fun actionTitle(): String?

    fun status(): String?

    fun companyInfo(): CompanyInfo?
}